spec/controllers/users_controller_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+  describe '#index' do
+    it 'redirects to home page' do
+      get :index
+      expect(response).to redirect_to(root_path)
+    end
+  end
+
+  let(:user){ create(:user) }
+  describe '#create' do
+    context 'with valid params' do
+      def valid_user
+        post :create, params: { user: attributes_for(:user)}
+      end
+
+      it 'saves a record to the database' do
+        count_before = User.count
+        valid_user
+
+        count_after = User.count
+        expect(count_after).to eq(count_before + 1)
+      end
+
+      it 'saves session id of current user' do
+        # u = create(:user)
+        # expect(session[:user_id]).to eq(u.id)
+
```

Comment:
Remove comment and space below it.

Probably better to set session[:user_id] to nil first and then do the comparison.  I like the way you did in the comment though.